Src/FluentAssertions/Common/TypeExtensions.cs

```diff
@@ -136,14 +113,10 @@ internal static Type[] GetClosedGenericInterfaces(Type type, Type openGenericTyp
                     .ToArray();
         }
 
-        public static bool IsComplexType(this Type type)
-        {
-            return HasProperties(type) && !AssertionOptions.IsValueType(type);
-        }
-
-        private static bool HasProperties(Type type)
+        public static bool OverridesEquals(this Type type)
         {
-            return type.GetProperties(PublicMembersFlag).Any();
+            MethodInfo methodInfo = type.GetMethod(nameof(object.Equals), new[]{ typeof(object)});
+            return (methodInfo != null) && (methodInfo.GetBaseDefinition().DeclaringType != methodInfo.DeclaringType);
```

Comment:
Does this check that `Equals(object)` also overrides `object.Equals(object)`? 
I.e. it is neither of
```
public bool Equals(object obj);
public new bool Equals(object obj);
```

Reply:
Only the 2nd is possible (I think)

Reply:
It fails for this corner case

```csharp
public class VirtualClass
{
	public virtual bool Equals(object obj) { ... }
}

public class VirtualClassOverride : VirtualClass
{
	public override bool Equals(object obj) { ... }
}
```

https://dotnetfiddle.net/qKKLbq

Reply:
I'll fix it.

Reply:
It should be enough to add an additional check that 
`methodInfo.GetBaseDefinition().DeclaringType == typeof(System.Object)`

Reply:
Took a bit more than that:

```
            IEnumerable<MethodInfo> methods = type
                .GetMethods(BindingFlags.Public | BindingFlags.Instance);

            methods = methods
                .Where(m => m.Name == "Equals" &&
                            m.GetParameters().Select(p => p.ParameterType).SequenceEqual(new[] {typeof(object)}));

            methods = methods
                .Where(m => m.GetBaseDefinition().DeclaringType != m.DeclaringType)
                .ToArray();

            return methods.Any();
```